src/planner/parabolic/HauserParabolicSmootherHelpers.cpp

```diff
@@ -0,0 +1,198 @@
+    , mInterpolator(mStateSpace)
+  {
+    // Do nothing
+  }
+
+  bool ConfigFeasible(const ParabolicRamp::Vector& x) override
+  {
+    Eigen::VectorXd eigX = toEigen(x);
+    auto state = mStateSpace->createState();
+    mStateSpace->expMap(eigX, state);
+    return mTestable->isSatisfied(state);
+  }
+
+  bool SegmentFeasible(
+      const ParabolicRamp::Vector& a, const ParabolicRamp::Vector& b) override
+  {
+    Eigen::VectorXd eigA = toEigen(a);
+    Eigen::VectorXd eigB = toEigen(b);
+
+    auto testState = mStateSpace->createState();
+    auto startState = mStateSpace->createState();
+    auto goalState = mStateSpace->createState();
+    mStateSpace->expMap(eigA, startState);
+    mStateSpace->expMap(eigB, goalState);
+
+    aikido::util::VanDerCorput vdc{1, false, false, mCheckResolution};
+
+    for (const auto alpha : vdc)
+    {
+      mInterpolator.interpolate(startState, goalState, alpha, testState);
+      if (!mTestable->isSatisfied(testState))
+      {
+        return false;
+      }
+    }
+    return true;
+  }
+
+private:
+  aikido::constraint::TestablePtr mTestable;
+  double mCheckResolution;
+  aikido::statespace::StateSpacePtr mStateSpace;
+  aikido::statespace::GeodesicInterpolator mInterpolator;
+};
+
+bool needsBlend(const ParabolicRamp::ParabolicRampND& rampNd)
+{
+  for (size_t idof = 0; idof < rampNd.dx1.size(); ++idof)
+  {
+    if (std::fabs(rampNd.dx1[idof]) > ParabolicRamp::EpsilonV)
+    {
+      return false;
+    }
+  }
+  return true;
+}
+
+bool tryBlend(
+    ParabolicRamp::DynamicPath& dynamicPath,
+    ParabolicRamp::RampFeasibilityChecker& feasibilityChecker,
+    int attempt,
+    double dtShortcut)
+{
+  // blending can completely remove waypoints from the trajectory in the case
+  // that two waypoints are closer than _blendRadius together - which means
+  // that waypoint indicies can change between iterations of the algorithm.
+  const size_t numRamps = dynamicPath.ramps.size();
+  const double tMax = dynamicPath.GetTotalTime();
+  double t = 0;
+
+  for (size_t iRamp = 0; iRamp < numRamps - 1; ++iRamp)
+  {
+    ParabolicRamp::ParabolicRampND& rampNd = dynamicPath.ramps[iRamp];
+    t += rampNd.endTime;
+
+    if (needsBlend(rampNd) && rampNd.blendAttempts == attempt)
+    {
+      const double t1 = std::max(t - dtShortcut, -ParabolicRamp::EpsilonT);
+      const double t2
+          = std::min(t + dtShortcut, tMax + ParabolicRamp::EpsilonT);
+
+      const bool success = dynamicPath.TryShortcut(t1, t2, feasibilityChecker);
+
+      if (success)
+      {
+        return true;
+      }
+      else
+      {
+        rampNd.blendAttempts++;
+      }
+    }
+  }
+
+  return false;
+}
+
+bool doShortcut(
+    ParabolicRamp::DynamicPath& dynamicPath,
+    aikido::constraint::TestablePtr testable,
+    double timelimit,
+    double checkResolution,
+    double tolerance,
+    aikido::util::RNG& rng)
+{
+  if (timelimit < 0.0)
+    throw std::runtime_error("Timelimit should be non-negative");
```

Comment:
Nits:
- Add a bounds check for `checkResolution`.
- Use `std::invalid_argument` instead of `std::runtime_error` for these exceptions.